### PR TITLE
Add pathPrefix option

### PR DESCRIPTION
**Pull request by ahmedelgabri**
I renamed the `paginatePost` to `pathPrefix` and made sure the returned URLs look like this

```
/ --> for the first page

/:num --> where :num > 1
OR
/:pathPrefix/:num --> where :num > 1
```

### CODE REVIEW

**Comment by Sorgrum** [comment]
Hey so funnily enough I was trying to do this exact same this and wrote almost identical code to your PR. But on further inspection it seems that the `paginatePost` argument is basically the `pathPrefix` argument your suggesting. I actually accomplished the same thing by using the existing `paginatePost` parameter and modifying the `createPaginatedPages` function like so:

```js
const createPaginatedPages = (posts, createPage, template, post) => {
  posts.forEach((group, index, groups) => {
    return createPage({
      path: buildPaginationRoute(getPageIndex(index), post),
      component: template,
      context: {
        group,
        pathPrefix: post,
        first: isFirstPage(index),
        last: isLastPage(index, groups),
        index: index + 1
      }
    });
  });
};
```
By passing in the pathPrefix in the context you can add an explicit check to see if there is one supplied and modify the link accordingly: 
```js
  const { group, index, first, last, pathPrefix } = pathContext;

  let previousUrl = index - 1 == 1 ? '' : (index - 1).toString();
  let nextUrl = (index + 1).toString();

  if (pathPrefix) {
    previousUrl = pathPrefix + "/" + previousUrl;
    nextUrl = pathPrefix + "/" + nextUrl;
  }
```
TL;DR I think the code for pathPrefix is already implemented, but maybe the naming is a little confusing. A smaller tweak to pass the pathPrefix into the context can be effective as well.

Also, its very possible I misinterpreted what `paginatePost` is used for. In that case some explanation would be very helpful.

**Comment by pixelstew** [comment]
HI @ahmedelgabri @Sorgrum 

I will have a look at this tonight - sorry for the delay in getting back. Work has been very busy.

**Comment by ahmedelgabri** [comment]
@Sorgrum you are right, I guess I got confused by the variable names `paginatePost` & `post`. I changed my PR. I renamed `paginatePost` to `pathPrefix` & the URLs generated will follow this pattern

```
/ --> for the first page

/:num --> where :num > 1
OR
/:pathPrefix/:num --> where :num > 1
```

and pagination logic will look like this

```js
  let previousUrl = index - 1 == 1 ? '' : index - 1
  let nextUrl = index + 1

  if (pathPrefix) {
    previousUrl = index - 1 === 1 ? '' : `${pathPrefix}/${previousUrl}`
    nextUrl = `${pathPrefix}/${nextUrl}`
  }
```

Take your time @pixelstew & thanks for the plugin 👍

**Comment by ahmedelgabri** [comment]
Actually I think a better way is to pass a function that gets used instead of `buildPaginationRoute` because in my case I wanted `/` & then `/page/2` etc... but I can see other people wanting `/page/1`, `/page/2` etc...

So passing a function with this signature might be more useful

```js
pathPrefix: 'page',
buildPath: (index, pathPrefix) => `${pathPrefix}/${index}`;

// or there is no need for pathPrefix anymore
buildPath: (index) => `page/${index}`;
```

if a user don't pass a function then `buildPaginationRoute` is used instead, what do you think?
